File: mobile_client/app/src/main/java/com/bessonov/musicappclient/api/ArtistAPI.java
package com.bessonov.musicappclient.api;

import com.bessonov.musicappclient.dto.ArtistInfoDTO;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.GET;
import retrofit2.http.POST;

public interface ArtistAPI {
    @GET("/artist/get/all")
    Call<List<ArtistInfoDTO>> getAll();

    @POST("/artist/get/byArtistId")
    Call<ArtistInfoDTO> getByAlbumId(@Body int artistId);

    @POST("/artist/get/byArtistId/list")
    Call<List<ArtistInfoDTO>> getByAlbumIdList(@Body List<Integer> artistIdList);
}

File: mobile_client/app/src/main/java/com/bessonov/musicappclient/api/ArtistAPI.kt
package com.bessonov.musicappclient.api

import com.bessonov.musicappclient.dto.ArtistInfoDTO
import com.bessonov.musicappclient.dto.UserArtistDTO
import com.bessonov.musicappclient.dto.UserRatingDTO
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Path

interface ArtistAPI {
    @GET("/artist/get/all")
    fun getAll(): Call<List<ArtistInfoDTO>>

    @GET("/artist/get/{artistId}")
    fun getByArtistId(@Path("artistId") artistId: Int): Call<ArtistInfoDTO>

    @POST("/artist/get/list")
    fun getByArtistIdList(@Body artistIdList: List<Int>): Call<List<ArtistInfoDTO>>

    @GET("/artist/get/user")
    fun getArtistUserList(): Call<List<ArtistInfoDTO>>

    @GET("/artist/add/{artistId}")
    fun addArtistToUserList(@Path("artistId") artistId: Int): Call<ArtistInfoDTO>

    @GET("/artist/remove/{artistId}")
    fun removeArtistFromUserList(@Path("artistId") artistId: Int): Call<ArtistInfoDTO>

    @POST("/artist/rate/{artistId}")
    fun rateArtist(@Path("artistId") artistId: Int, @Body rateId: Int): Call<ArtistInfoDTO>
}
